Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM python:3.9.5-slim-buster as base
+
+RUN pip install flask
+RUN pip install poetry
+
+COPY *.toml . 
+
+RUN poetry install
```

Comment:
You could move this into the individual stages and have it vary slightly.

E.g. `poetry install --no-dev` for production and `poetry install` for development. That way certain packages which are only for development can be skipped in the production image.